caddyhttp/fastcgi/fastcgi.go

```diff
@@ -258,13 +258,19 @@ func (h Handler) buildEnv(r *http.Request, rule Rule, fpath string) (map[string]
 		env[envVar[0]] = replacer.Replace(envVar[1])
 	}
 
-	// Add all HTTP headers to env variables
+	// Add all HTTP headers (except Caddy-Rewrite-Original-URI ) to env variables
 	for field, val := range r.Header {
-		header := strings.ToUpper(field)
-		header = headerNameReplacer.Replace(header)
-		env["HTTP_"+header] = strings.Join(val, ", ")
+		
+		if strings.ToUpper(field) != strings.ToUpper(internalRewriteFieldName) {
+		   header := strings.ToUpper(field)
+		   header = headerNameReplacer.Replace(header)
+		   env["HTTP_"+header] = strings.Join(val, ", ")
+		} 
```

Comment:
`go fmt` will fix the indentation here too.

Also, let's do this "fail fast" style, so that instead of indenting all the logic of the loop, we simply do:

```go
if strings.ToLower(field) == strings.ToLower(internalRewriteFieldName) {
    continue
}
...
```